app/src/main/java/edu/uncc/inclass11/GradesFragment.java

```diff
@@ -189,6 +191,14 @@ void setCourse_number(String course_number) {
         void setCourse_name(String course_name) {
             TextView textView = view.findViewById(R.id.textViewCourseName);
             textView.setText(course_name);
+
+            ImageView imageViewDelete = view.findViewById(R.id.imageViewDelete);
+            imageViewDelete.setOnClickListener(view -> firebaseFirestore.collection("grades").document(course_name)
+                    .delete()
+                    .addOnSuccessListener(unused -> Log.d("demo", "Grade successfully deleted"))
+                    .addOnFailureListener(e -> Log.w("demo", "Error deleting grade", e)));
+
+            mListener.goGrades();
```

Comment:
This needs to be moved inside of the success listener. Putting this here would, I imagine, create an infinite loop of loading the grades fragment. Unless it pops the back stack, in which case it'll just kick back to the login screen (I think).